cloud/misc/ovirt.py

```diff
@@ -199,231 +285,944 @@
+                    )
+            else:
+                return custom_script
+
+    def _cloud_init_network(self):
+        """
+        :rtype: ovirtsdk.xml.params.NetworkConfiguration
+        """
+        nic_name = self.module.params['nic_name']
+        static_ip_address = self.module.params['static_ip_address']
+        static_netmask = self.module.params['static_netmask']
+        static_gateway = self.module.params['static_gateway']
+
+        if all([nic_name, static_ip_address, static_netmask, static_gateway]):
+            return params.NetworkConfiguration(
+                nics=params.Nics(nic=[
+                    params.NIC(
+                        name=nic_name,
+                        boot_protocol="STATIC",
+                        on_boot=True,
+                        network=params.Network(
+                            ip=params.IP(
+                                address=static_ip_address,
+                                netmask=static_netmask,
+                                gateway=static_gateway
+                            )
+                        )
+                    )
+                ])
+            )
+
+    def _cloud_init_user_groups(self):
+        user_groups = self.module.params['authorized_key_user_groups']
+        if user_groups is not None:
+            return params.Groups(
+                group=[params.Group(name=group.strip()) for group in user_groups.lower().strip().split(',')]
+            )
+
+    def _cloud_init_user(self):
+        authorized_key_user = self.module.params['authorized_key_user']
+        if authorized_key_user is not None:
+            return params.Users(
+                user=[
+                    params.User(
+                        user_name=authorized_key_user,
+                        groups=self._cloud_init_user_groups()
+                    )
+                ]
+            )
+
+    def _generate_cloud_init(self):
+        """
+        :rtype : ovirtsdk.xml.params.Initialization
+        """
+
+        authorized_key_user = self.module.params['authorized_key_user']
+        instance_host_name = self.module.params['instance_host_name']
+
+        script = self._get_custom_script()
+        key = self._get_auth_key()
+        network = self._cloud_init_network()
+
+        files = None
+        auth_key = None
+
+        if script is not None:
+            files = params.Files(
+                file=[params.File(name="/tmp/_vmcustomscript", content=script, type_="PLAINTEXT")]
+            )
+
+        if key is not None:
+            auth_key = params.AuthorizedKeys(
+                authorized_key=[
+                    params.AuthorizedKey(
+                        user=params.User(user_name=authorized_key_user),
+                        key=key
+                    ),
+                ]
+            )
+
+        cloud_init = params.CloudInit(
+            host=instance_host_name,
```

Comment:
You have to create a host instance with the attribute "address" for that, like so:

        if instance_host_name is not None:
            host = params.Host(
                address=instance_host_name,
            )
(...)
        cloud_init = params.CloudInit(
            host=host,
            files=files,
(...)